packages/react-dom/src/FadeIn.tsx

```diff
@@ -0,0 +1,34 @@
+import { type CSSProperties, type ComponentPropsWithoutRef, type ElementType } from 'react'
+import { useInView } from './useInView'
+
+type FadInBaseProps = {
+  delay?: number
+  duration?: number
+  timingFunction?: CSSProperties['animationTimingFunction']
+}
+type FadeInProps<TAs extends ElementType> = {
+  as?: TAs
+} & Omit<ComponentPropsWithoutRef<TAs>, 'as' | keyof FadInBaseProps> &
+  FadInBaseProps
+
+export function FadeIn<TAs extends ElementType = 'div'>({
+  as,
+  delay = 0,
+  duration = 200,
+  timingFunction = 'linear',
+  ...rest
+}: FadeInProps<TAs>) {
+  const Component = as || 'div'
+  const { inView, ref } = useInView()
```

Comment:
Component name "FadeIn" doesn't hint at observer usage - maybe we need an interface to control the observer?

Reply:
Oh, I agree. We can expose InViewOptions as FadeInProps

Reply:
I added inViewOptions as one prop of FadeInProps